sno-plus/README.md

```diff
@@ -0,0 +1,203 @@
+- we indicate in the bootstrap in place the disk to use for installation
+- we set network type to OVN
+
+```
+apiVersion: v1
+baseDomain: karmalabs.com
+compute:
+- name: worker
+  replicas: 0
+controlPlane:
+  name: master
+  replicas: 1
+metadata:
+  name: mycluster
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 10.0.0.0/16
+  clusterNetwork:
+  - cidr: 10.132.0.0/14
+    hostPrefix: 23
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  none: {}
+BootstrapInPlace:
+  InstallationDisk: /dev/vda
+pullSecret: 'XXX'
+sshKey: |
+  ssh-rsa ZZZ
+```
+
+With this install config, we then generate an ignition to use for SNO
+
+```
+mkdir ocp
+cp install-config.yaml ocp
+openshift-install --dir=ocp create single-node-ignition-config
+```
+
+Provided this content is stored in /root/config.ign, we inject it from a rhcos live iso using the following invocation
+
+```
+coreos-installer install --ignition=/root/config.ign /dev/vda
+```
+
+After a while, we should see an output similar to the following one:
+
+```
+[root@u08.dev5.kni.lab.eng.bos.redhat.com ~]# oc get node
+NAME                          STATUS   ROLES           AGE     VERSION
+mycluster-sno.karmalabs.com   Ready    master,worker   176m    v1.22.3+b93fd35
+```
+
+
+2. Adding additional workers
+
+Although ignition for workers is not generated when running openshift-install as in the previous section, this can be retrieved using curl
+
+`curl 192.168.122.235:22624/config/worker > worker.ign`
+
+To add more workers to the SNO, we craft an openshift iso using coreos-installer and making sure it installs the following target ignition, which contains two relevant entries
+
+- the source pointing to the final ignition to use
+- an overwrite of /etc/hosts so that api.$cluster.$domain and api-int.$cluster.$domain point to the ip of the SNO
+
+```
+{
+    "ignition": {
+        "config": {
+           "merge": [
+              {
+              "source": "http://192.168.122.235:22624/config/worker"
+              }
+           ]
+        },
+    "version": "3.1.0"
+    },
+    "storage": {
+        "files": [
+            {
+                "contents": {
+                    "source": "data:text/plain;charset=utf-8;base64,MTI3LjAuMC4xICAgbG9jYWxob3N0IGxvY2FsaG9zdC5sb2NhbGRvbWFpbiBsb2NhbGhvc3Q0IGxvY2FsaG9zdDQubG9jYWxkb21haW40Cjo6MSAgICAgICAgIGxvY2FsaG9zdCBsb2NhbGhvc3QubG9jYWxkb21haW4gbG9jYWxob3N0NiBsb2NhbGhvc3Q2LmxvY2FsZG9tYWluNgoxOTIuMTY4LjEyMi4yMzUgYXBpLWludC5teWNsdXN0ZXIua2FybWFsYWJzLmNvbSBhcGkubXljbHVzdGVyLmthcm1hbGFicy5jb20=",
+                    "verification": {}
+                },
+                "filesystem": "root",
+                "mode": 420,
+                "overwrite": true,
+                "path": "/etc/hosts"
+            }
+        ]
+    }
+}
+```
+
+Provided this content is stored in /root/config.ign, we inject it from a rhcos live iso using the following invocation
+
+```
+coreos-installer install --ignition=/root/config.ign /dev/vda
```

Comment:
again, this is confusing. Is this booted from the live iso and the config.ign copied over?